Sources/PackageGraph/PackageGraphLoader.swift

```diff
@@ -211,6 +211,12 @@ private func createResolvedPackages(
             .filter({ $0.product.type != .test })
         let productDependencyMap = productDependencies.createDictionary({ ($0.product.name, $0) })
 
+        // Establish if there are multiple products with the same name.
+        let duplicateProductNames = Set(packageBuilder.products.map { $0.product.name }.findDuplicates())
```

Comment:
Do we really require `Set()` here? I don't think findDuplicates should return duplicate elements.

Reply:
This won't look for the duplicate products across packages, we should probably do that.